inc/common.php

```diff
@@ -1312,66 +1312,81 @@ function saveWikiText($id, $text, $summary, $minor = false) {
+        'newRevision'    => 0,      // only available in the after hook
+        'newContent'     => $text,
+        'oldContent'     => $currentContent,
+        'summary'        => $summary,
+        'contentChanged' => (bool)($text != $currentContent), // confirm later
+        'changeInfo'     => '',     // set prior to event
+        'changeType'     => null,   // set prior to event
+        'sizechange'     => strlen($text) - strlen($currentContent), // TBD
+    );
+    // determin change type and relevant elements of event data
     if ($REV) {
-        $svdta['changeType'] = DOKU_CHANGE_TYPE_REVERT;
-        $svdta['changeInfo'] = $REV;
-    } elseif (!file_exists($svdta['file'])) {
-        $svdta['changeType'] = DOKU_CHANGE_TYPE_CREATE;
+        $data['changeType'] = DOKU_CHANGE_TYPE_REVERT;
+        $data['changeInfo'] = $REV;
+    } elseif (!file_exists($pagefile)) {
+        $data['changeType'] = DOKU_CHANGE_TYPE_CREATE;
     } elseif (trim($text) == '') {
         // empty or whitespace only content deletes
-        $svdta['changeType'] = DOKU_CHANGE_TYPE_DELETE;
+        $data['changeType'] = DOKU_CHANGE_TYPE_DELETE;
         // autoset summary on deletion
-        if (blank($svdta['summary'])) {
-            $svdta['summary'] = $lang['deleted'];
+        if (blank($data['summary'])) {
+            $data['summary'] = $lang['deleted'];
         }
     } elseif ($minor && $conf['useacl'] && $INPUT->server->str('REMOTE_USER')) {
         //minor edits only for logged in users
-        $svdta['changeType'] = DOKU_CHANGE_TYPE_MINOR_EDIT;
+        $data['changeType'] = DOKU_CHANGE_TYPE_MINOR_EDIT;
+    } else {
+        $data['changeType'] = DOKU_CHANGE_TYPE_EDIT;
     }
 
-    $event = new Event('COMMON_WIKIPAGE_SAVE', $svdta);
+    $event = new Event('COMMON_WIKIPAGE_SAVE', $data);
     if (!$event->advise_before()) return;
 
     // if the content has not been changed, no save happens (plugins may override this)
-    if (!$svdta['contentChanged']) return;
+    if (!$data['contentChanged']) return;
 
-    // register external edit, creation or deletion as a regular changelog entry
-    detectExternalEdit($id);
-
-    if (
-        ($svdta['changeType'] == DOKU_CHANGE_TYPE_CREATE) ||
-        ($svdta['changeType'] == DOKU_CHANGE_TYPE_REVERT && !file_exists($svdta['file']))
-    ) {
+    // confirm again event data that may altered by event handlers
+    // plugin may alter the value of oldRevision to last revision to ignore extrnal edit
+    if ($data['oldRevision'] === false) {
+        // current file was not existed, no external edit occured
         $filesize_old = 0;
+    } elseif ($data['oldRevision'] == $currentRevision) {
+        // add current external revision entry to changelog as a regular changelog entry
+        detectExternalEdit($id);
+        $filesize_old = $currentSize;
+//  } elseif ($data['oldRevision'] == $lastRevision) {
+//      $filesize_old = strlen(rawWiki($id, $data['oldRevision']));
```

Comment:
commented code remaining

Reply:
@Klap-in , I have updated saveWikiText(). please see also FIXME comment!